src/libs/blueprint/conduit_blueprint_mesh_utils.cpp

```diff
@@ -345,6 +345,217 @@ find_domain_id(const Node &node)
     return domain_id;
 }
 
+//---------------------------------------------------------------------------
+// @brief Slice the n_src array using the indices stored in ids. We use the
+//        array classes for their [] operators that deal with interleaved
+//        and non-interleaved arrays.
+template <typename T, typename IndexType>
+inline void
+typed_slice_array(const T &src, const std::vector<IndexType> &ids, T &dest)
+{
+    size_t n = ids.size();
+    for(size_t i = 0; i < n; i++)
+        dest[i] = src[ids[i]];
+}
+
+//---------------------------------------------------------------------------
+/**
+ @brief Slice a node as an array using its native data type.
+ */
+template <typename IndexType>
+void
+slice_array_internal(const conduit::Node &n_src_values,
+                     const std::vector<IndexType> &ids,
+                     Node &n_dest_values)
+{
+    // Copy the DataType of the input conduit::Node but override the number of elements
+    // before copying it in so assigning to n_dest_values triggers a memory
+    // allocation.
+    auto dt = n_src_values.dtype();
+    n_dest_values = DataType(n_src_values.dtype().id(), ids.size());
+
+    // Do the slice.
+    if(dt.is_int8())
+    {
+        auto dest(n_dest_values.as_int8_array());
+        typed_slice_array(n_src_values.as_int8_array(), ids, dest);
+    }
+    else if(dt.is_int16())
+    {
+        auto dest(n_dest_values.as_int16_array());
+        typed_slice_array(n_src_values.as_int16_array(), ids, dest);
+    }
+    else if(dt.is_int32())
+    {
+        auto dest(n_dest_values.as_int32_array());
+        typed_slice_array(n_src_values.as_int32_array(), ids, dest);
+    }
+    else if(dt.is_int64())
+    {
+        auto dest(n_dest_values.as_int64_array());
+        typed_slice_array(n_src_values.as_int64_array(), ids, dest);
+    }
+    else if(dt.is_uint8())
+    {
+        auto dest(n_dest_values.as_uint8_array());
+        typed_slice_array(n_src_values.as_uint8_array(), ids, dest);
+    }
+    else if(dt.is_uint16())
+    {
+        auto dest(n_dest_values.as_uint16_array());
+        typed_slice_array(n_src_values.as_uint16_array(), ids, dest);
+    }
+    else if(dt.is_uint32())
+    {
+        auto dest(n_dest_values.as_uint32_array());
+        typed_slice_array(n_src_values.as_uint32_array(), ids, dest);
+    }
+    else if(dt.is_uint64())
```

Comment:
should we add float32 and float64 cases for completeness?

Reply:
I added those cases.